### ISSUE TITLE
Behavior of F.dropout in eval mode

### ISSUE THREAD

**Issue by sidml** [pytorch/pytorch#26338]
## 📚 Documentation
The documentation for F.dropout should probably mention that putting the model in eval mode doesn't disable dropout.

A small example
```python
import torch
import torch.nn as nn
import torch.nn.functional as F

class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()

    def forward(self, x):
        x = F.dropout(x)
        return x

net = Net()                                                             

a = torch.ones(10)                                                     

net(a)                                                                  
tensor([0., 0., 0., 2., 2., 2., 0., 0., 2., 2.]) 

net.eval()

net(a)
tensor([0., 0., 0., 2., 2., 2., 0., 0., 2., 2.])


```

If the model is set in eval mode, the dropout layer is not disabled.
is_training should take care of this.. But it may be better if this behavior is explicitly mentioned in the docs.

**Comment by soumith** [comment]
it makes sense to mention it for the functional version, yes. `nn.Dropout` module will disable it.

**Comment by SsnL** [comment]
`F.dropout` is functional. It is actually expected that you need to manually specify the arg `training=...` to turn it on/off.

**Comment by yenkunyuan-hue** [comment]
Another dropout function, nn.Dropout(), which can automatically switch, which isnot functional:

class Net(nn.Module):
def __init__(self,p=0.2):
        super(Net, self).__init__()
        self.drop = nn.Dropout(p=p)
    def forward(self, x):
        x = self.drop(x)
        return x

Here is the output:
tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
tensor([1.2500, 1.2500, 1.2500, 1.2500, 1.2500, 1.2500, 1.2500, 1.2500, 1.2500,
        0.0000])
tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])

If you want to use 2D input, try nn.Dropout2d instead
